### PR TITLE
[BPK-2385] Fix button icon alignment and style

### PR DESCRIPTION
**Pull request by tiagoengel**
This adjusts all styling issues we had with the button and aligns it to the RN version.

Here are some screenshots:

## Overlap with RN
Screenshot 2019-04-12 at 15 30 06
Screenshot 2019-04-12 at 15 30 16
## Overlap Primary/secondary
Screenshot 2019-04-12 at 16 14 57

## Icon alignment
Screenshot_1555082192

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/master/CONTRIBUTING.md)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

**Comment by georgegillams** [comment]
Thank you for your contribution.
You can use the following dependency to test your application while we review your pull request:

`com.github.Skyscanner:backpack-android:715f19fe6a54d44d661fe9d9812ec4ea46882ba2`

File: app/src/main/res/layout/fragment_button_outline.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
     android:id="@+id/btn_default"
     android:layout_width="wrap_content"
     android:layout_height="wrap_content"
-    android:text="Default"
+    android:text="button"
```
Comment (amolgupta) [inline app/src/main/res/layout/fragment_button_outline.xml]: minor, Upper case `B` in `button`

File: Backpack/src/main/java/net/skyscanner/backpack/button/TextMeasurement.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  fun getTextWidth(text: String): Int {
+    if (textBoundsRect == null) {
+      textBoundsRect = Rect()
+    }
+    val tokenized = tokenize(text)
+    paint.getTextBounds(tokenized, 0, tokenized.length, textBoundsRect)
+    return textBoundsRect!!.width()
+  }
+
+  private fun tokenize(text: String): String {
```
Comment (amolgupta) [inline Backpack/src/main/java/net/skyscanner/backpack/button/TextMeasurement.kt]: can we add a test case for this?

File: Backpack/src/main/java/net/skyscanner/backpack/button/BpkButton.kt [hunk 3]
```diff
@@ -333,3 +434,35 @@ private fun darken(@ColorInt normalColor: Int, factor: Float = .2f): Int {
   hsv[2] *= 1f - factor // value component
   return Color.HSVToColor(hsv)
 }
+
+private fun convertToBitmap(drawable: Drawable): Bitmap? {
```
Comment (amolgupta) [inline Backpack/src/main/java/net/skyscanner/backpack/button/BpkButton.kt]: this can be in some util file, so other components can use it if required.

Comment (tiagoengel) [reply Backpack/src/main/java/net/skyscanner/backpack/button/BpkButton.kt]: I'd keep here for now and move only when we actually need it for other components. Rule of three :)

File: Backpack/src/main/java/net/skyscanner/backpack/button/TextMeasurement.kt (hunk 2 above)
Comment (tiagoengel) [reply Backpack/src/main/java/net/skyscanner/backpack/button/TextMeasurement.kt]: Sure, I'll do

File: app/src/main/res/layout/fragment_button_outline.xml (hunk 1 above)
Comment (tiagoengel) [reply app/src/main/res/layout/fragment_button_outline.xml]: Yes, version is wrong from the previous release

**Comment by tiagoengel** [comment]
Yes the `package-lock` had the wrong version from the previous release

**Comment by georgegillams** [comment]
Thank you for your contribution.
You can use the following dependency to test your application while we review your pull request:

`com.github.Skyscanner:backpack-android:65a2e7a5d2099d057fc1dc8900f536eafe9cbb55`

**Comment by georgegillams** [comment]
Thank you for your contribution.
You can use the following dependency to test your application while we review your pull request:

`com.github.Skyscanner:backpack-android:ff684d9d910f8457083309f002134d25d4fa5c8e`

**Pull request merged**